chat-plugins/panagrams-beta.js

```diff
@@ -152,4 +152,127 @@ exports.commands = {
+				var guess = dex[toId(message)];
+				for (var i in Game.users) {
+					if (Game.users[i].latestIp === user.latestIp) return user.sendTo(this, 'One of your alts have already guessed once in this game of panagram.');
+				}
+				if (toId(guess.species) in Game.guessed) return user.sendTo(this, 'That Pokemon has already been guessed.');
+				if (toId(guess.species) === toId(Game.answer.species)) {
+					this.add('|html|<b>' + user.name + '</b> guessed <b>' + guess.species + '</b>, which was the correct answer! Congratulations!');
+					Game.end();
+				} else if (guess.num < 1 || guess.baseSpecies) user.sendTo(this, guess.species + ' is either an alternate form or doesn\'t exist in the games. These cannot be guessed.');
+				else {
+					if (Object.keys(Game.users).length < 3) {
+						this.add('|html|<b>' + user.name + '</b> guessed <b>' + guess.species + '</b>, but was not the correct answer...');
+						Game.guessed[toId(guess.species)] = user.userid;
+					} else {
+						this.add('|html|<b>' + user.name + '</b> guessed <b>' + guess.species + '</b>, but was not the correct answer. You have failed to guess the Pokemon, which was <b>' + (answer.name || answer.species) + '</b>');
+						Game.end();
+					}
+				}
+			} else {
+				if (message && message !== true) {
+					this.add('|c|' + user.getIdentity(this.id) + '|' + message);
+				}
+			}
+			this.update();
+		};
+	}
+	Panagram.prototype.end = function (forced) {
+		if (forced) this.room.add('The game of panagram has been forcibly ended.');
+		if (this.sessions > 1 && !forced) pGames[this.room.id] = new Panagram(this.room, --this.sessions);
+		else {
+			this.room.chat = Rooms.Room.prototype.chat;
+			delete pGames[this.room.id];
+		}
+	};
+	return Panagram;
+})();
+
+exports.commands = {
+	panagramrules: 'panagramhelp',
+	phelp: 'panagramhelp',
+	panagramhelp: function (target, room, user) {
+		if (!this.canBroadcast()) return;
+		this.sendReplyBox('<b>Panagram Commands</b><br>' +
+			'<li>/panagram <em>Category</em> - Starts a game of panagram that follows the specified category. Valid categories include <i>Pokemon, Items, Abilities</i> and <i>moves</i>. Not specifying a category starts a pokemon panagram by default. Requires + or higher to use.' +
+			'<li>/panagram end <small>or</small> /endp - Ends a game of panagram. Requires + or higher to use.<br>' +
+			'Users can guess answers by simply typing them into the chat.'
+		);
+	},
+
+	panagram: function (target, room, user, connection, cmd) {
+		if (pGames[room.id]) return this.errorReply("There is currently a game of panagram going on in this room.");
+		if (!this.can('roomban', null, room)) return this.errorReply("You must be ranked @ or higher to start a game of panagram in this room.");
+		if (room.id !== 'gamechamber') return this.errorReply('|html|You can only start a game of Panagram in the <button name = "send" value = "/join gamechamber">Game Chamber</button>');
+		if (!target || isNaN(target)) target = 1;
```

Comment:
Silent return?  Bad.

We won't do games of one 9/10 - make this:
``this.errorReply("Usage: /panagram [number of games to start]");``